certbot/tests/auth_handler_test.py

```diff
@@ -231,6 +213,21 @@ def test_perform_failure(self):
         self.assertRaises(
             errors.AuthorizationError, self.handler.handle_authorizations, mock_order)
 
+    def test_max_retries_exceeded(self):
+        authzrs = [gen_dom_authzr(domain="0", challs=acme_util.CHALLENGES)]
+        mock_order = mock.MagicMock(authorizations=authzrs)
+
+        # We will return STATUS_PENDING twice before returning STATUS_VALID.
+        self.mock_net.poll.side_effect = _gen_mock_on_poll(retry=2)
+
+        with self.assertRaises(errors.AuthorizationError) as error:
+            # We retry only once, so retries will be exhausted before STATUS_VALID is returned.
+            self.handler.handle_authorizations(mock_order, False, 1)
+        self.assertTrue('All challenges could not be checked on time' in str(error.exception))
+
+        self.assertRaises(
+            errors.AuthorizationError, self.handler.handle_authorizations, mock_order, False, 1)
```

Comment:
I don't understand why we're testing this again. Is there something I'm missing?

Reply:
No, I think it is something from a previous implementation of this test. I remove it.